src/msg/async/ProtocolV1.cc

```diff
@@ -350,21 +350,17 @@ void ProtocolV1::write_event() {
     connection->write_lock.unlock();
 
     // if r > 0 mean data still lefted, so no need _try_send.
-    if (r == 0) {
+    if (r == 0 && ack_left) {
```

Comment:
What if ``ack_left != 0`` but ``is_queued() == true`` at that point?

Reply:
if `is_queued() == true` then it's because `out_q` is not empty, but that does not matter for the `_try_send()` function.